cmd/gardener-metrics-exporter/app.go

```diff
@@ -122,30 +125,37 @@ func run(o *options, closeCh chan os.Signal) error {
 	return nil
 }
 
-func setupInformerFactories(kubeconfigPath string, stopCh <-chan struct{}) (gardeninformers.SharedInformerFactory, kubeinformers.SharedInformerFactory, error) {
-	k8sGardenClient, err := kubernetes.NewClientFromFile(kubeconfigPath)
+func newConfigFromBytes(kubeconfig string) *restclient.Config {
+	kubecf, err := ioutil.ReadFile(kubeconfig)
 	if err != nil {
-		return nil, nil, err
+		fmt.Println("Could not read config")
+		os.Exit(1)
 	}
+	configObj, err := clientcmd.Load(kubecf)
+	if err != nil {
+		fmt.Println("Could not load config")
+		os.Exit(1)
+	}
+	clientConfig := clientcmd.NewDefaultClientConfig(*configObj, &clientcmd.ConfigOverrides{})
+	client, err := clientConfig.ClientConfig()
+	if err != nil {
+		return nil
+	}
+	return client
+}
 
-	gardenerClient := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
-		&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeconfigPath},
-		&clientcmd.ConfigOverrides{},
-	)
-
-	gardenerClientConfig, err := gardenerClient.ClientConfig()
+func setupInformerFactories(kubeconfigPath string, stopCh <-chan struct{}) (gardeninformers.SharedInformerFactory, kubeinformers.SharedInformerFactory, error) {
+	k8sClient, err := kubernetes.NewForConfig(newConfigFromBytes(kubeconfigPath))
```

Comment:
Let's store the result of `newConfigFromBytes(kubeconfigPath)` in a variable and pass it to the `NewForConfig()` functions. Then we have to create the config only once. And we could also return an error in `newConfigFromBytes` (see comment in the function).